pkg/persistence/team_assets_filter.go

```
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2020 Appvia Ltd <info@appvia.io>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package persistence
+
+import "time"
+
+var (
+	// TeamAssetFilters provides a set of useful filters for cost listing
+	TeamAssetFilters TeamAssetFilterFuncs
+)
+
+// TeamAssetFilterFunc is a function to filter the cost list
+type TeamAssetFilterFunc func(*TeamAssetListOptions)
+
+// TeamAssetListOptions is used to specify the restrictions on the returned assets or costs
+type TeamAssetListOptions struct {
+	From            *time.Time
+	To              *time.Time
+	BillingMonth    *uint8
+	BillingYear     *uint16
+	Provider        string
+	Account         string
+	TeamIdentifier  string
+	AssetIdentifier string
+	WithDeleted     bool
+}
+
+// NewTeamAssetListOptions returns a cost list options
+func NewTeamAssetListOptions() *TeamAssetListOptions {
+	return &TeamAssetListOptions{
+		WithDeleted: false,
```

This is the default empty value, so no need to define it.

---

Yeah - I realised this, but I wanted to be explicit about the intention - so it's clear to anyone who looks at this code that the intention is to hide deleted by default (rather than by accident). 

Helps poor souls wondering in future why there's no deleted records in their query (poor souls like I was when the deleted rows weren't there because gorm silently hides rows with a column called deleted_at if that column is populated on a row...)